<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>882</width>
    <height>698</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="toolTip">
   <string/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>881</width>
      <height>661</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab1">
     <attribute name="title">
      <string>Лабиринт</string>
     </attribute>
     <widget class="QWidget" name="mazeWidget" native="true">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>20</y>
        <width>861</width>
        <height>601</height>
       </rect>
      </property>
      <widget class="QPushButton" name="openMazeButton">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>0</y>
         <width>231</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Открыть файл с лабиринтом</string>
       </property>
      </widget>
      <widget class="QPushButton" name="saveMazeButton">
       <property name="geometry">
        <rect>
         <x>270</x>
         <y>0</y>
         <width>231</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Сохранить лабринт в файл</string>
       </property>
      </widget>
      <widget class="QPushButton" name="generateMazeButton">
       <property name="geometry">
        <rect>
         <x>610</x>
         <y>140</y>
         <width>221</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Сгенерировать лабиринт</string>
       </property>
      </widget>
      <widget class="QLabel" name="rowsLabel">
       <property name="geometry">
        <rect>
         <x>580</x>
         <y>50</y>
         <width>161</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Количество строк</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="rowsCount">
       <property name="geometry">
        <rect>
         <x>750</x>
         <y>50</y>
         <width>111</width>
         <height>31</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>50</number>
       </property>
      </widget>
      <widget class="QPushButton" name="findPathButton">
       <property name="geometry">
        <rect>
         <x>610</x>
         <y>190</y>
         <width>221</width>
         <height>31</height>
        </rect>
       </property>
       <property name="toolTip">
        <string/>
       </property>
       <property name="text">
        <string>Найти путь между точками</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="colsCount">
       <property name="geometry">
        <rect>
         <x>750</x>
         <y>90</y>
         <width>111</width>
         <height>31</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>50</number>
       </property>
      </widget>
      <widget class="QLabel" name="colsLabel">
       <property name="geometry">
        <rect>
         <x>580</x>
         <y>90</y>
         <width>161</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Количество столбцов</string>
       </property>
      </widget>
      <widget class="QPushButton" name="clearPointsButton">
       <property name="geometry">
        <rect>
         <x>610</x>
         <y>230</y>
         <width>221</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Сбросить выбранные точки</string>
       </property>
      </widget>
      <widget class="QGraphicsView" name="mazeView">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>550</width>
         <height>550</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="clearAllMazeSceneButton">
       <property name="geometry">
        <rect>
         <x>610</x>
         <y>280</y>
         <width>221</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Сбросить всё</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab2">
     <attribute name="title">
      <string>Пещера</string>
     </attribute>
     <widget class="QWidget" name="caveWidget" native="true">
      <property name="geometry">
       <rect>
        <x>9</x>
        <y>9</y>
        <width>861</width>
        <height>601</height>
       </rect>
      </property>
      <widget class="QSpinBox" name="limitLifeCount">
       <property name="geometry">
        <rect>
         <x>720</x>
         <y>190</y>
         <width>111</width>
         <height>31</height>
        </rect>
       </property>
       <property name="maximum">
        <number>7</number>
       </property>
      </widget>
      <widget class="QPushButton" name="initCaveButton">
       <property name="geometry">
        <rect>
         <x>550</x>
         <y>150</y>
         <width>221</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Начальная инициализация</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="limitDeathCount">
       <property name="geometry">
        <rect>
         <x>720</x>
         <y>230</y>
         <width>111</width>
         <height>31</height>
        </rect>
       </property>
       <property name="maximum">
        <number>7</number>
       </property>
      </widget>
      <widget class="QLabel" name="chanceLabel">
       <property name="geometry">
        <rect>
         <x>550</x>
         <y>100</y>
         <width>161</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Шанс на начальную инициализацию</string>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLabel" name="lifeLabel">
       <property name="geometry">
        <rect>
         <x>550</x>
         <y>190</y>
         <width>161</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Предел рождения</string>
       </property>
      </widget>
      <widget class="QLabel" name="deathLabel">
       <property name="geometry">
        <rect>
         <x>550</x>
         <y>230</y>
         <width>161</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Предел смерти</string>
       </property>
      </widget>
      <widget class="QPushButton" name="saveCaveButton">
       <property name="geometry">
        <rect>
         <x>270</x>
         <y>20</y>
         <width>231</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Сохранить пещеру в файл</string>
       </property>
      </widget>
      <widget class="QPushButton" name="openCaveButton">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>231</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Открыть файл с пещерой</string>
       </property>
      </widget>
      <widget class="QGraphicsView" name="caveView">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>511</width>
         <height>511</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="clearAllCaveSceneButton">
       <property name="geometry">
        <rect>
         <x>550</x>
         <y>470</y>
         <width>221</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Сбросить всё</string>
       </property>
      </widget>
      <widget class="QPushButton" name="startAutoGenerateCaveButton">
       <property name="geometry">
        <rect>
         <x>550</x>
         <y>370</y>
         <width>221</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Автоматическая генерация</string>
       </property>
      </widget>
      <widget class="QLabel" name="timeLabel">
       <property name="geometry">
        <rect>
         <x>550</x>
         <y>320</y>
         <width>161</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Тайминг автогенерации в мс</string>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QSpinBox" name="timeCount">
       <property name="geometry">
        <rect>
         <x>720</x>
         <y>320</y>
         <width>111</width>
         <height>31</height>
        </rect>
       </property>
       <property name="maximum">
        <number>3000</number>
       </property>
      </widget>
      <widget class="QLabel" name="colsLabel_2">
       <property name="geometry">
        <rect>
         <x>550</x>
         <y>60</y>
         <width>161</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Количество столбцов</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="colsCount_2">
       <property name="geometry">
        <rect>
         <x>720</x>
         <y>60</y>
         <width>111</width>
         <height>31</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>50</number>
       </property>
      </widget>
      <widget class="QLabel" name="rowsLabel_2">
       <property name="geometry">
        <rect>
         <x>550</x>
         <y>20</y>
         <width>161</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Количество строк</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="rowsCount_2">
       <property name="geometry">
        <rect>
         <x>720</x>
         <y>20</y>
         <width>111</width>
         <height>31</height>
        </rect>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>50</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="chanceInitCount">
       <property name="geometry">
        <rect>
         <x>720</x>
         <y>100</y>
         <width>111</width>
         <height>31</height>
        </rect>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
      </widget>
      <widget class="QPushButton" name="stopAutoGenerateCaveButton">
       <property name="geometry">
        <rect>
         <x>550</x>
         <y>410</y>
         <width>221</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Остановить генерацию</string>
       </property>
      </widget>
      <widget class="QPushButton" name="oneGenerateCaveButton">
       <property name="geometry">
        <rect>
         <x>550</x>
         <y>270</y>
         <width>221</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Один шаг</string>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>882</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
